Add CLI endpoint to validate input files

nilsvu opened the pull request:
## Proposed changes

Try it like this:

```
spectre validate INPUT_FILE
```

In particular, this will be used in #4996 to validate input files before scheduling runs.

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

nilsvu committed:
Add validate input file to CLI

knelli2 commented on tools/ValidateInputFile.py:
```diff
@@ -0,0 +1,151 @@
+    catch the exception and print the 'render_context()' as well.
+
+    Attributes:
+      input_file_path: Path to the input file on disk.
+      lineno: Line number in the input file where the parse error occurred.
```
I think `line_number` is clearer and it's not that much longer.

knelli2 reviewed (changes requested):
I think you can drop the build dir commit because it's already taken care of in #5023. Then I believe these can be merged simultaneously rather than having to rebase one.

knelli2 reviewed (changes requested):
This test timed out a couple times, maybe because it takes a sec to get going? Probably should increase test timeout if possible

knelli2 commented:
Looks good, but I think you used black to format instead of yapf because of your other PR.

nilsvu committed:
Install Py packages on macOS CI with pip

nilsvu committed:
Upgrade rich version

The older version was only used because it's in Spack v0.18, but we
don't need to rely on that anymore.

nilsvu commented:
@kidder @wthrowe can one of you take this one please?
